rotkehlchen/data/importer.py

```diff
@@ -784,6 +784,74 @@ def import_blockfi_trades_csv(self, filepath: Path) -> Tuple[bool, str]:
                     return False, str(e)
         return True, ''
 
+    def _consume_binance_purchases(self, csv_row: Dict[str, Any]) -> None:
+        """
+        Consume the file containing only purchases by credit/debit card from Binance.
+        """
+        # TODO perhaps put the 'Completed' into a variable as it may change
+        if csv_row['Status'] != 'Completed':
+            log.debug(f'Ignoring not completed Binance purchase. {csv_row}')
+            return
+
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date(UTC)'],
+            formatstr='%Y-%m-%d %H:%M:%S',
+            location='Binance',
+        )
+
+        amount, base_asset = csv_row['Final Amount'].split(' ')
+        base_asset = symbol_to_asset_or_token(base_asset)
+        amount = deserialize_asset_amount(amount)
+        quote_amount, quote_asset = csv_row['Amount'].split(' ')
+        quote_asset = symbol_to_asset_or_token(quote_asset)
+        quote_amount = deserialize_asset_amount(quote_amount)
+        fee, fee_currency = csv_row['Fees'].split(' ')
+        fee = deserialize_fee(fee)
+        fee_currency = symbol_to_asset_or_token(fee_currency)
+        rate = Price((quote_amount + fee) / amount)
```

Comment:
For this refer to the comment I made here: https://github.com/rotki/rotki/pull/2979#pullrequestreview-671162734